### DISCUSSION TITLE
TiTiler as an AWS Serverless Application (SAM)

### DISCUSSION THREAD

**Discussion by vincentsarago** [developmentseed/titiler#313] [Show and tell]
Using https://github.com/lambgeo/titiler-layer (custom AWS Lambda layer built with a custom GDAL build) I've created an AWS SAM (Serverless Application Model) application which is publicly available over https://serverlessrepo.aws.amazon.com/applications/us-east-1/552819999234/TiTiler

This enable almost a `one click` deployment solution

<p><a href="https://console.aws.amazon.com/lambda/home?#/create/app?applicationId=arn:aws:serverlessrepo:us-east-1:552819999234:applications/TiTiler" rel="noreferrer"><img src="https://cdn.rawgit.com/buildkite/cloudformation-launch-stack-button-svg/master/launch-stack.svg" alt="Launch Stack"></a></p>

**Comment by vincentsarago** [comment]
FYI: I just updated the SAM application to the latest official TiTiler version (0.7.1)

<p><a href="https://console.aws.amazon.com/lambda/home?#/create/app?applicationId=arn:aws:serverlessrepo:us-east-1:552819999234:applications/TiTiler" rel="noreferrer"><img src="https://cdn.rawgit.com/buildkite/cloudformation-launch-stack-button-svg/master/launch-stack.svg" alt="Launch Stack"></a></p>

**Reply by mradamcox** [reply]
When I try to deploy, I get `User: arn:aws:iam::###2:user/### is not authorized to perform: serverlessrepo:GetApplication on resource: arn:aws:serverlessrepo:us-east-1:552819999234:applications/TiTiler`. I'm totally new to Lambda, so is there something else I need to do?

**Reply by vincentsarago** [reply]
@mradamcox I just tested with my account and It seems to work well, maybe you don't have enough permission with your AWS User to perform `serverlessrepo:GetApplication`. If you are using a company AWS account, reach out to your admin to update your role... if not I'm not sure what's going on

**Reply by mradamcox** [reply]
I did have to add some permissions to my IAM user. Seems like there wasn't a pre-made policy for `serverlessrepo` actions so I created a policy with the following and added it.
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "serverlessrepo:*",
            "Resource": "*"
        }
    ]
}
```
I also had to add the `AmazonAPIGatewayAdministrator` policy. My user had a lot of permissions already, so this isn't an exhaustive list of what you need, but just wanted to mention this in case it's helpful for others.
